### PR TITLE
Using Eno-Webservices

### PR DESCRIPTION
**Pull request by loichenninger**

### CODE REVIEW

File: src/main/java/fr/insee/pogues/api/remote/eno/transforms/EnoClientImpl.java [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestTemplate;
+
+@Service
+public class EnoClientImpl implements EnoClient{
+	
+	private static final Logger logger = LogManager.getLogger(EnoClientImpl.class);
+	
+	@Autowired
+	RestTemplate restTemplate;
```
Comment (MichaelC67) [inline src/main/java/fr/insee/pogues/api/remote/eno/transforms/EnoClientImpl.java:37]: consider adding visibility to the class variables
```suggestion
	RestTemplate restTemplate;
```

Comment (MichaelC67) [inline src/main/java/fr/insee/pogues/api/remote/eno/transforms/EnoClientImpl.java:37]: Consider adding visibility to class variables

File: src/main/java/fr/insee/pogues/api/remote/eno/transforms/EnoClientImpl.java [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+	
+	@Override
+	public void getParameters () throws Exception{
+		
+		URIBuilder uriBuilder = new URIBuilder();
+		uriBuilder.setScheme(enoScheme).setHost(enoHost).setPath("/parameter/default");
+		
+	    RestTemplate restTemplate = new RestTemplate();
+	    ResponseEntity<String> result = restTemplate.exchange(uriBuilder.build(), HttpMethod.GET, null, String.class);
+	    System.out.println(result.getBody());
```
Comment (MichaelC67) [inline src/main/java/fr/insee/pogues/api/remote/eno/transforms/EnoClientImpl.java:120]: Remove System.out.println ?

File: src/main/java/fr/insee/pogues/transforms/DDI32ToDDI33Impl.java [hunk 3]
```diff
@@ -9,19 +9,19 @@
...
 
 
 @Service
 public class DDI32ToDDI33Impl implements DDI32ToDDI33 {
 	
-final static Logger logger = LogManager.getLogger(DDI32ToDDI33Impl.class);
+	final static Logger logger = LogManager.getLogger(DDI32ToDDI33Impl.class);
+	
+	@Autowired
+	EnoClient enoClient;
```
Comment (MichaelC67) [inline src/main/java/fr/insee/pogues/transforms/DDI32ToDDI33Impl.java:24]: Visibility ?

File: src/main/java/fr/insee/pogues/transforms/DDIToODTImpl.java [hunk 4]
```diff
@@ -9,16 +9,16 @@
...
-import fr.insee.eno.postprocessing.NoopPostprocessor;
-import fr.insee.eno.postprocessing.Postprocessor;
-import fr.insee.eno.preprocessing.DDIPreprocessor;
+import fr.insee.pogues.api.remote.eno.transforms.EnoClient;
 
 @Service
 public class DDIToODTImpl implements DDIToODT {
+	
+	@Autowired
+	EnoClient enoClient;
```
Comment (MichaelC67) [inline src/main/java/fr/insee/pogues/transforms/DDIToODTImpl.java:21]: visibility ?

File: src/main/java/fr/insee/pogues/transforms/DDIToPDFImpl.java [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import fr.insee.pogues.api.remote.eno.transforms.EnoClient;
+
+@Service
+public class DDIToPDFImpl implements DDIToPDF{
+	
+	@Autowired
+	EnoClient enoClient;
```
Comment (MichaelC67) [inline src/main/java/fr/insee/pogues/transforms/DDIToPDFImpl.java:21]: Visibility?

File: src/main/java/fr/insee/pogues/transforms/DDIToXFormImpl.java [hunk 6]
```diff
@@ -1,21 +1,22 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import fr.insee.pogues.api.remote.eno.transforms.EnoClient;
+
 @Service
 public class DDIToXFormImpl implements DDIToXForm {
+	
+	@Autowired
+	EnoClient enoClient;
```
Comment (MichaelC67) [inline src/main/java/fr/insee/pogues/transforms/DDIToXFormImpl.java:19]: Visibility ?

File: src/main/resources/env/dev/pogues-bo.properties [hunk 7]
```diff
@@ -40,6 +40,8 @@ fr.insee.pogues.permission.ldap.permission.name = ou
 fr.insee.pogues.role.admin = Administrateur_Pogues
 fr.insee.pogues.role.user = Utilisateurs_Pogues
 # Metadata service
-fr.insee.pogues.api.remote.metadata.url = http://dvrmspogbolht01.ad.insee.intra/ddi-access-services/api
+fr.insee.pogues.api.remote.metadata.url = http://localhost:8080/ddi-access-services-2/api
 # Stromae service
-fr.insee.pogues.api.remote.stromae.vis.url = http://dvstromaeldb01.ad.insee.intra:8080/exist/restxq/visualize
\ No newline at end of file
+fr.insee.pogues.api.remote.stromae.vis.url = http://dvstromaeldb01.ad.insee.intra:8080/exist/restxq/visualize
```
Comment (MichaelC67) [inline src/main/resources/env/dev/pogues-bo.properties:45]: Anonymization useful here ?  with *** ?

File: src/main/resources/env/dev/pogues-bo.properties [hunk 8]
```diff
@@ -40,6 +40,8 @@ fr.insee.pogues.permission.ldap.permission.name = ou
...
+# Eno service
+fr.insee.pogues.api.remote.eno.host = dvenowslht01.ad.insee.intra
```
Comment (MichaelC67) [inline src/main/resources/env/dev/pogues-bo.properties:47]: Anonymization useful here ? with *** ?

File: src/main/java/fr/insee/pogues/transforms/FOToPDFImpl.java [hunk 9]
```diff
@@ -26,12 +26,14 @@
...
-import fr.insee.eno.Constants;
-
 @Service
 public class FOToPDFImpl implements FOToPDF {
 
 	final static Logger logger = LogManager.getLogger(FOToPDFImpl.class);
+	
+	public static final String TEMP_FOLDER_PATH = System.getProperty("java.io.tmpdir") + "/eno";
+	
+	public static final String FINAL_PDF_EXTENSION = "-final-out.fo";
```
Comment (MichaelC67) [inline src/main/java/fr/insee/pogues/transforms/FOToPDFImpl.java:36]: Strange variable name ? PDF_EXTENSION for a fo file ?

File: src/main/java/fr/insee/pogues/api/remote/eno/transforms/EnoClientImpl.java [hunk 10]
```diff
@@ -0,0 +1,139 @@
...
+	@Override
+	public String getDDIToFO(File fileInput) throws URISyntaxException, ClientProtocolException, IOException {
+		HttpEntity entityResponse = callEnoApi(fileInput, "/questionnaire/DEFAULT/fo");
+        return EntityUtils.toString(entityResponse, "UTF-8");
+	}
+	
+	@Override
+	public String getDDITOLunaticXML(File fileInput) throws URISyntaxException, ClientProtocolException, IOException {
+		HttpEntity entityResponse = callEnoApi(fileInput, "/questionnaire/DEFAULT/lunatic-xml");
+        return EntityUtils.toString(entityResponse, "UTF-8");
```
Comment (alicela) [inline src/main/java/fr/insee/pogues/api/remote/eno/transforms/EnoClientImpl.java:97]: Add a contant for "UTF-8" and "/questionnaire/DEFAULT/"

**Pull request merged**